a5py/routines/runmixin.py

```diff
@@ -492,8 +495,13 @@ def getwall_loads(self, weights=True):
             Angle of incidence (TODO).
         """
         self._require("_endstate")
-        ids, energy, weight = self.getstate("walltile", "ekin", "weight",
+        if particle_filter is None:
+            ids, energy, weight = self.getstate("walltile", "ekin", "weight",
                                             state="end", endcond="wall")
+        else:
+            ids, energy, weight = self.getstate("walltile", "ekin", "weight",
+                                            state="end", endcond="wall",
+                                            ids = particle_filter)
```

Comment:
No need to separate these cases since by default `getstate(ids=None)` i.e. you can use `getstate(ids=ids)`